src/components/structures/RoomView.js

```diff
@@ -534,30 +560,19 @@ module.exports = React.createClass({
 
     // set off a pagination request.
     onMessageListFillRequest: function(backwards) {
-        if (!backwards)
+        var dir = backwards ? EventTimeline.BACKWARDS : EventTimeline.FORWARDS;
+        if(!this._timelineWindow.canPaginate(dir)) {
```

Comment:
ugh, this component is so bloody complicated, it's impossible to know what you need to null-guard when.

In this case: if there is no `_timelineWindow`, render won't create the  `ScrollView`, so we can't get `onMessageListFillRequests`. I think.